### PR TITLE
Harsen/tap 2360 develop auto recovery

### PR DESCRIPTION
**Pull request by HarsenLin**

### CODE REVIEW

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_blocker_violations**, ActualValue: **3**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **57**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=harsen/TAP-2360-develop-AutoRecovery&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **0.0**
- Status: **ERROR**, MetricKey: **new_blocker_violations**, ActualValue: **3**
- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **57**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=harsen/TAP-2360-develop-AutoRecovery&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_major_violations**, ActualValue: **1**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=harsen/TAP-2360-develop-AutoRecovery&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **57.77777777777778**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=harsen/TAP-2360-develop-AutoRecovery&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **57.77777777777778**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=harsen/TAP-2360-develop-AutoRecovery&id=tapdata

File: iengine/api/src/test/java/io/tapdata/inspect/AutoRecoveryTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    private static final String taskId = "test-task-id";
+    private static final String inspectTaskId = "test-inspect-task-id";
+    private static final long interval = 50;
+
+    @Test
+    void testDuplicateServer() {
+        DuplicateAutoRecoveryException recoveryException = null;
+        try (
+            AutoRecovery ignore1 = AutoRecovery.init(taskId);
+            AutoRecovery ignore2 = AutoRecovery.init(taskId)
```
Comment (ply0011) [inline iengine/api/src/test/java/io/tapdata/inspect/AutoRecoveryTest.java:36]: 这块可以用更好的断言
DuplicateAutoRecoveryException e = assertThrows(DuplicateAutoRecoveryException.class, ()->AutoRecovery.init(taskId));
assertEquals("xxx", e.getMessage());
类似这样写，单测不建议用try catch，可能会把一些错误掩盖

**Review by ply0011 — CHANGES_REQUESTED** [review]

File: iengine/api/src/test/java/io/tapdata/inspect/AutoRecoveryTest.java (hunk 1 above)
Comment (HarsenLin) [reply iengine/api/src/test/java/io/tapdata/inspect/AutoRecoveryTest.java]: 很好的建议，已调整

**Review by ply0011 — APPROVED** [review]

**Pull request merged**